constants/secret.go

```diff
@@ -20,4 +20,7 @@ const (
 
 	// SecretLogMask defines the secret mask to be used when distributing logs that contain secrets.
 	SecretLogMask = "***"
+
+	// SecretRestrictedCharacters defines the set of characters that a secret name cannot contain.
```

Comment:
Can we note what this character is in the human form? It's not clear if you don't know what the character means. Google is always helpful but it would be nice to just state it human friendly.

Reply:
fair callout

Reply:
@GregoryDosh do you want to add a commit to this PR with that info? i think it would be worthwhile to call out what `\x00` is